frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -156,8 +140,14 @@ const CreateSnapshotForm = withHandlePromise<SnapshotResourceProps>((props) => {
   const [pvcObj, setPVCObj] = React.useState<PersistentVolumeClaimKind>(null);
   const [snapshotName, setSnapshotName] = React.useState(`${pvcName || 'pvc'}-snapshot`);
   const [snapshotClassName, setSnapshotClassName] = React.useState('');
+  const [scObj, setSCObj] = React.useState<StorageClassResourceKind>(null);
```

Comment:
IMO there's too many stuff stored in state, which dont actually need to be stored -that is `pvcObj` and `scObj`.

You already have `pvcName` from which you can calculate `pvcObj`.
same for  `scObj` for which you already have `snapshotClassName`.

Reply:
I would suggest keeping both for the following reasons: 
the scObj is the storageclass object which  is calculated from the scList 
`  const currentSC = scList.find((sc) => sc.metadata.name === currentPVC?.spec?.storageClassName);`

We cannot use snapshotClassName to calculate it and it required to continuously calculate our filterWrapper func hence we would like to look for its change.

the pvcObj on the other hand is used in countless places alongside pvcName to render various UI forms and items. It is not a change i have made and hence in light of keeping things unbroken and avoiding major refactors, i would like to keep it for now atleast :)

Reply:
I understand you need the object, but you dont need to store it in state.

for `scObj` you can just find it before the first usage - which in this case is

```
const snapshotClassFilter = React.useCallback(
    (snapshotClass: VolumeSnapshotClassKind) => scObj?.provisioner.includes(snapshotClass?.driver),
    [scObj],
  );
```
would be 

```
const scObj = scList.find((sc) => sc.metadata.name === pvcObj?.spec?.storageClassName);
  const snapshotClassFilter = React.useCallback(
    (snapshotClass: VolumeSnapshotClassKind) => scObj?.provisioner.includes(snapshotClass?.driver),
    [scObj],
  );
```

result is one less state and less args in React.useEffect.


Same can be done for pvcObj, but I agree it doesnt have to be addressed in this PR since its already there.

Reply:
And you do not even need whole scObj, you just use the provisioner from sc

```
const scProvisioner = scList.find((sc) => sc.metadata.name === pvcObj?.spec?.storageClassName)?.provisioner;
  const snapshotClassFilter = React.useCallback(
    (snapshotClass: VolumeSnapshotClassKind) => scProvisioner.includes(snapshotClass?.driver),
    [scProvisioner],
  );
```

same for useEffect.

Which further improves the dependencies and number of rerenders - now we depend on simple string, not object or list of objects